Sources/Datadog/Datadog.swift

```diff
@@ -288,31 +288,36 @@ public class Datadog {
         self.launchTimeProvider = launchTimeProvider
     }
 
-    /// Internal feature made only for tests purpose.
-    static func deinitializeOrThrow() throws {
-        guard Datadog.instance != nil else {
-            throw ProgrammerError(description: "Attempted to stop SDK before it was initialized.")
-        }
-
-        // First, reset internal loggers:
+#if DD_SDK_COMPILED_FOR_TESTING
+    /// Flushes all authorised data for each feature, tears down and deinitializes the SDK.
+    /// - It flushes all data authorised for each feature by performing its arbitrary upload (without retrying).
+    /// - It completes all pending asynchronous work in each feature.
+    ///
+    /// This is highly experimental API and only supported in tests.
+    public static func flushAndDeinitialize() {
```

Comment:
alternatively, you can keep this method internal and call it from outside via notifications
```swift
init() {
  NotificationCenter.default.addObserver("DD.flush", self, #selector(flushAndDeinit))
}
// in tests
NotificationCenter.postNotif("DD.flush")
```

Reply:
Notifications are made for broadcasting. Direct method call is IMO way cleaner and easier to read. 

In both cases, we would need to use conditional compilation for this experimental `deinit()` code - not only from the the public interface, but also for internals sanity.